viewer/viewer.js

```diff
@@ -4514,6 +4517,136 @@ app.get('/:nodeName/:id/bodypng/:bodyType/:bodyNum/:bodyName', checkProxyRequest
   });
 });
 
+/**
+ * Get a file given a hash of that file
+ */
+
+app.get('/huntfile/hash/:hash', logAction('huntfile'), function(req, res) {
+
+  var hash = null;
+  var nodeName = null;
+  var sessionID = null;
+
+  buildSessionQuery(req, function(bsqErr, query, indices) {
+    if (bsqErr) {
+      res.status(400);
+      return res.end(bsqErr);
+    }
+    if (Config.debug) {
+      console.log(`sessions.json ${indices} query`, JSON.stringify(query, null, 1));
+    }
+
+    Db.searchPrimary(indices, 'session', query, function(err, sessions) {
+      if (err ) {
+        console.log ("Error -> Db Search ", err);
+        res.status(400);
+        res.end(err);
+      } else if (sessions.error) {
+        console.log ("Error -> Db Search ", sessions.error);
+        res.status(400);
+        res.end(sessions.error);
+      } else {
+          if (Config.debug) {
+            console.log("sessions.json result", util.inspect(sessions, false, 50));
+          }
+          if (sessions.hits.hits.length > 0) {
+
+            nodeName = sessions.hits.hits[0]._source.node;
+            sessionID = sessions.hits.hits[0]._id;
```

Comment:
ok, then I would over ride the query size to 1, since you only need 1 and I would also over ride the sort to be most recent since that one is most likely still to have the pcap.

Reply:
Sounds good. It will reduce the response time as well. I will make the necessary changes.

Reply:
I did over ride the sort, size, and _source fields as follows:
query.size = 1;
query.sort = { lastPacket: { order: 'desc' } };
 query._source = ["node"];